conan/tools/gnu/mingw.py

```diff
@@ -0,0 +1,12 @@
+def is_mingw(conanfile):
+    """
+    Validate if current compiler in host setttings is related to MinGW
+    :param conanfile: ConanFile instance
+    :return: True, if the host compiler is related to MinGW, otherwise False.
+    """
+    host_os = conanfile.settings.get_safe("os")
+    host_compiler = conanfile.settings.get_safe("compiler")
+    is_mingw_gcc = host_os == "Windows" and host_compiler == "gcc"
+    is_mingw_clang = host_os == "Windows" and host_compiler == "clang" and \
+                     not conanfile.settings.get_safe("compiler.runtime")
```

Comment:
it's worth noting that I've prefered this condition over `conanfile.settings.get_safe("compiler.libcxx")` because libcxx is removed in `configure()` of pure C recipes, so it's not a reliable compiler attribute to disambiguate between clang mingw and clang-cl.